asserts/tool/sign_test.go

```diff
@@ -0,0 +1,331 @@
+}
+
+func (s *signSuite) TestSignKeyIDFlatYAMLRevisionWithinHeaders(c *C) {
+	req := tool.SignRequest{
+		KeyID:       s.testKeyID,
+		AuthorityID: "user-id1",
+
+		AssertionType:      "model",
+		StatementMediaType: "application/x-yaml",
+		Statement:          []byte(flatModelYaml + "revision: 12"),
+	}
+
+	assertText, err := tool.Sign(&req, s.keypairMgr)
+	c.Assert(err, IsNil)
+
+	a, err := asserts.Decode(assertText)
+	c.Assert(err, IsNil)
+
+	c.Check(a.Type(), Equals, asserts.ModelType)
+	c.Check(a.Revision(), Equals, 12)
+
+	expectedHeaders := expectedModelHeaders()
+	expectedHeaders["revision"] = "12"
+
+	c.Check(a.Headers(), DeepEquals, expectedHeaders)
+
+	c.Check(a.Body(), IsNil)
+}
+
+func headersForJSON() map[string]interface{} {
+	return map[string]interface{}{
+		"series":         "16",
+		"brand-id":       "user-id1",
+		"model":          "baz-3000",
+		"core":           "core",
+		"architecture":   "amd64",
+		"gadget":         "brand-gadget",
+		"kernel":         "baz-linux",
+		"store":          "brand-store",
+		"allowed-modes":  nil,
+		"required-snaps": []string{"foo", "bar"},
+		"class":          "fixed",
+		"extra-flag":     true,
+		"timestamp":      "2015-11-25T20:00:00Z",
+	}
+}
+
+func (s *signSuite) TestSignKeyIDFlatJSONRevisionWithinHeaders(c *C) {
+	hdrs := headersForJSON()
+	hdrs["revision"] = 12
+	statement, err := json.Marshal(hdrs)
+	c.Assert(err, IsNil)
+
+	req := tool.SignRequest{
+		KeyID:       s.testKeyID,
+		AuthorityID: "user-id1",
+
+		AssertionType:      "model",
+		StatementMediaType: "application/json",
+		Statement:          statement,
+	}
+
+	assertText, err := tool.Sign(&req, s.keypairMgr)
+	c.Assert(err, IsNil)
+
+	a, err := asserts.Decode(assertText)
+	c.Assert(err, IsNil)
+
+	c.Check(a.Type(), Equals, asserts.ModelType)
+	c.Check(a.Revision(), Equals, 12)
+
+	expectedHeaders := expectedModelHeaders()
+	expectedHeaders["revision"] = "12"
+
+	c.Check(a.Headers(), DeepEquals, expectedHeaders)
+
+	c.Check(a.Body(), IsNil)
+}
+
+func (s *signSuite) TestSignKeyIDNestedJSONWithBodyAndRevision(c *C) {
+	hdrs := headersForJSON()
+	statement, err := json.Marshal(map[string]interface{}{
+		"headers":      hdrs,
+		"content-body": "CONTENT",
+	})
+	c.Assert(err, IsNil)
+
+	req := tool.SignRequest{
+		KeyID:       s.testKeyID,
+		AuthorityID: "user-id1",
+
+		AssertionType:      "model",
+		StatementMediaType: "application/json",
+		Statement:          statement,
+
+		Revision: 11,
+	}
+
+	assertText, err := tool.Sign(&req, s.keypairMgr)
+	c.Assert(err, IsNil)
+
+	a, err := asserts.Decode(assertText)
+	c.Assert(err, IsNil)
+
+	c.Check(a.Type(), Equals, asserts.ModelType)
+	c.Check(a.Revision(), Equals, 11)
+
+	expectedHeaders := expectedModelHeaders()
+	expectedHeaders["revision"] = "11"
+	expectedHeaders["body-length"] = "7"
+
+	c.Check(a.Headers(), DeepEquals, expectedHeaders)
+
+	c.Check(a.Body(), DeepEquals, []byte("CONTENT"))
+}
+
+func (s *signSuite) TestSignAccountKeyHandleNestedYAML(c *C) {
```

Comment:
Is the "NestedYAML" part of the name relevant? It's seems like an implementation detail of the test to me.